packages/block-editor/src/components/link-control/index.js

```diff
@@ -315,28 +495,56 @@ function LinkControl( {
 			ref={ wrapperNode }
 			className="block-editor-link-control"
 		>
-			{ isEditingLink || ! value ? (
+			{ isResolvingLink && (
+				<div className="block-editor-link-control__loading">
+					<Spinner /> { __( 'Creating' ) }…
+				</div>
+			) }
+
+			{ ( isEditingLink || ! value ) && ! isResolvingLink && (
 				<LinkControlSearchInput
 					value={ inputValue }
 					onChange={ onInputChange }
-					onSelect={ ( suggestion ) => {
-						onChange( { ...value, ...suggestion } );
-						stopEditing();
+					onSelect={ async ( suggestion ) => {
+						if (
+							suggestion.type &&
+							CREATE_TYPE === suggestion.type
+						) {
```

Comment:
A lot of code duplication between this and the `onClick` in the rendered `LinkControlSearchCreate` above. Seems hard to maintain and easy to fall out of sync.

Reply:
If there are no objects I think we can address this in a follow up PR. I'll create an Issue.

Reply:
This has been consolidated into a single handler method.